tests/test_chimera.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+    tests/test_chimera.cpp -- This demonstrates a hybrid usage of pybind11, where the
```

Comment:
Casters for types not wrapped with `py::class_` are perfectly normal: `chrono.h`, `complex.h`, `eigen.h`, `functional.h`, `numpy.h`, `stl.h` and probably many more in client code.

I think it's nice to have this test, even though it's more an end-to-end test than a unit test, but it's important to explain here, for example starting with:

This original purpose of this test is to exercise that `type_caster<std::reference_wrapper<T>>` works for sub-casters that only have a `const T&` operator, but not a `T&` operator. ... continue with explanation why this is not a more focused unit test ...

Reply:
Done.